lib/view_component/test_helpers.rb

```diff
@@ -73,12 +73,15 @@ def with_controller_class(klass)
 
     def with_request_url(path)
       old_request_path_parameters = request.path_parameters
+      old_request_query_parameters = request.query_parameters
       old_controller = defined?(@controller) && @controller
 
       request.path_parameters = Rails.application.routes.recognize_path(path)
+      request.set_header("action_dispatch.request.query_parameters", request.send(:parse_query, path.split("?")[1]))
```

Comment:
Did you look into a way to pull this off with a public API? I'd really prefer we avoid using `send`.

Reply:
I agree 100% :+1:. The other day I didn't find anything and hoped there was a way to set query parameters similar to `#path_parameters=` that most people knew about.

I'll have another look :+1:.